app/services/jwt_token.rb

```diff
@@ -34,8 +34,20 @@ def to_s
   end
 
   # This method does not verify if the token signature is valid
+  # defined? keyword checks if expression is currently defined. It
+  # therefore, helps to memoize the nil JWT values.
   def decoded_payload
-    @decoded_payload ||= JWT.decode(token, nil, false).first
+    return @decoded_payload if defined?(@decoded_payload)
+    @decoded_payload = JWT.decode(token, nil, false).first
+
+    unless @decoded_payload.is_a?(Hash)
+      @decoded_payload = nil
+      logger.error "Invalid decoded JWT format."
```

Comment:
perhaps also add `logger.debug "received payload: #{@decodded_payload}` and move setting it to `nil` to after the logging (just so it's easier to debug what failed)